numpy/linalg/tests/test_linalg.py

```diff
@@ -1221,6 +1221,13 @@ class _TestNormBase:
     dt = None
     dec = None
 
+    @staticmethod
+    def check_dtype(a, res):
+        if issubclass(a.dtype.type, np.inexact):
+            assert_equal(res.dtype, a.real.dtype)
+        else:
+            assert_(issubclass(res.dtype.type, np.floating))
```

Comment:
Note that this `else` branch is redundant here, as `np.floating` is a subclass of `np.inexact`.

Reply:
I think this was clearer the way it was before

Reply:
> I think this was clearer the way it was here before @BvB93's suggestion

Yeah, in light of https://github.com/numpy/numpy/pull/17709#discussion_r518008120 I would agree with this.